e2e/specs/st_set_page_config.spec.js

```diff
@@ -29,12 +29,4 @@ describe("st.set_page_config", () => {
   it("sets the page title", () => {
     cy.title().should("eq", "Heya, world? · Streamlit");
   });
-
-  it("collapses the sidebar", () => {
```

Comment:
The answer is kinda? Yes, these are more difficult (without adding specific attributes that exhibit this), BUT props are created in the styled components, so we can test in Jest and get the same result (ie whether a prop is set or not vs whether a class is added or not), so that reduces Cypress times! 😉 If we wanted to include image snapshots, then Cypress would remain a worthy use case.

You can see in https://github.com/streamlit/streamlit/pull/2314/files for `frontend/src/components/core/Sidebar/Sidebar.test.tsx` that the tests in Sidebar leverage the props in jest. This ends up saving time in Cypress and perform the same operation.

Reply:
Mm... the Sidebar.test.tsx unit tests don't provide the same level of assurance that e2e tests would in terms of catching regressions (as is always the case for unit tests vs e2e tests). Specifically, the Jest test would not detect any issues from Python code call to where the Sidebar is instantiated.

If we're removing the e2e attribute detection, and there's no longer a good verification method in e2e, I think a screenshot diff would be a reasonable replacement. A TODO Github ticket in lieu of that is also fine!

Sidenote: my overall impression (~70% confidence) is that our Cypress times are mostly slow due to switching between e2e test files, not because of the number of tests in each file. That is, each e2e file has a fixed cost to spin up the Streamlit server that dominates the cost of actually running each test.

Reply:
Yea. That's fair. The easiest solution would be putting a data attribute of the state of the sidebar that we could check (versus a class).